query/query_cache.rs

```diff
@@ -7,33 +7,39 @@
 use std::{
     hash::{DefaultHasher, Hash, Hasher},
     sync::{
-        atomic::{AtomicU64, Ordering},
         Arc,
+        atomic::{AtomicU64, Ordering},
     },
 };
+use std::sync::Mutex;
+
+use moka::sync::Cache;
+use tracing::{event, Level};
+use typeql::query::stage::reduce::Stat;
 
 use compiler::executable::pipeline::ExecutablePipeline;
+use concept::thing::statistics::Statistics;
 use ir::{
     pipeline::{fetch::FetchObject, function::Function},
     translation::pipeline::TranslatedStage,
 };
-use moka::sync::Cache;
 use resource::{
-    constants::database::{QUERY_PLAN_CACHE_FLUSH_STATISTICS_CHANGE_PERCENT, QUERY_PLAN_CACHE_SIZE},
+    constants::database::{QUERY_PLAN_CACHE_FLUSH_ANY_STATISTIC_CHANGE_FRACTION, QUERY_PLAN_CACHE_SIZE},
     perf_counters::QUERY_CACHE_FLUSH,
 };
+use storage::sequence_number::SequenceNumber;
 use structural_equality::StructuralEquality;
 
 #[derive(Debug)]
 pub struct QueryCache {
     cache: Cache<IRQuery, ExecutablePipeline>,
-    statistics_size: AtomicU64,
+    last_statistics: Mutex<Statistics>,
```

Comment:
We have to keep the last used statistics the query cache is operating against, so we can calculate a long-term diff. This isn't really much  more expensive than what we were doing before, since we cloned the entire Statistics on every update anyway.